packages/dataviews/src/types.ts

```diff
@@ -94,6 +94,11 @@ export type Field< Item > = {
 	 */
 	Edit?: ComponentType< DataFormControlProps< Item > >;
 
+	/**
+	 * Optional config for editing the field.
+	 */
+	editAs?: 'radio';
```

Comment:
I do think we need a config like that. Maybe I would name it "control". 

I think this also means that like "field types" we need a folder of "controls" and we already have three of them (NumberControl, TextControl and SelectControl). I believe these should be treated similarly to RadioControl.

Reply:
Given there's a `Edit` prop to provide a custom edit function, how about `editControl` so it's connected by the prefix?

My thinking is that "controls" seem too open-ended: it seems you can list any control and it'll be used. I see it more as a "edit configuration" if that makes sense.

Reply:
The question I'm wondering though is whether this should be part of the `field` definition or the `form` object.

Reply:
It is "edit" configuration yes, but I don't like the name "editAs" personally. I think the question also is whether we see folks registering new "controls" without necessarily changing the "type" of the field. And the question for me seems "yes" given that the "type" controls more than just "edit", I want to use the "integer" sorting/validation... but I want to build a custom control for my field.

Reply:
How about changing the `Edit` signature to also support a predefined set of controls?

Reply:
I reworked this to overload `Edit`, this was my [alternative option](https://github.com/WordPress/gutenberg/pull/64370#discussion_r1709223705) as well.